app/services/ocean_service.rb

```diff
@@ -0,0 +1,62 @@
+class OceanService
+  ISLAND_SCATTER_RADIUS = 200
+  CENTER = [500,500]
+
+  def initialize(number_of_islands)
+    @number_of_islands = number_of_islands
+  end
+
+  def call
+    { islands: islands }
+  end
+
+  def islands
+    if @number_of_islands.zero? || @number_of_islands.nil?
+      []
+    else
+      size = island_size
+      biggest_island_size = size.sort.last
+      generate_positions(biggest_island_size).each_with_index.map { |position,i|IslandService.new(position, size[1-i]).call}
+    end
+  end
+
+  def distance(edge)
+    pt1 = edge.first
+    pt2 = edge.last
+    mx = (pt2.first - pt1.first)
+    my = (pt2.last - pt1.last)
+    Math.sqrt(mx*mx + my*my)
+  end
+
+  def generate_positions(biggest_island_size)
+    if @number_of_islands == 1
+      [make_point]
+    elsif @number_of_islands == 2
+      begin
+        points = [make_point,make_point]
+      end while (2*biggest_island_size) > distance(points)
+      points
+    else
+      polygon = Geometry::Polygon.regular_polygon(CENTER, ISLAND_SCATTER_RADIUS, @number_of_islands)
+      polygon.make_irregular
+    end
+  end
+
+  def make_point
+    d = 150 #minimum distance from canvas edge
+    d2 = 850 # minimumdistance from canvas edge
+    point = [rand(d..d2),rand(d..d2)]
+  end
+
+  def island_size
+    min = 30
+    sizes = []
+    until sizes.sum > 300
+      @number_of_islands.times do |n|
+          max2 = (150 - 20*n)
+          sizes << rand(min..max2)
+        end
+      end
+    sizes
```

Comment:
interessting... what does this do ? ... seems complicated for something that returns randoms